internal/controller/postgrescluster/instance.go

```diff
@@ -1197,7 +1197,7 @@ func addPGBackRestToInstancePodSpec(cluster *v1beta1.PostgresCluster,
 		}
 	}
 	if !dedicatedRepoEnabled {
-		if err := pgbackrest.AddRepoVolumesToPod(cluster, template,
+		if err := pgbackrest.AddRepoVolumesToPod(cluster, template, repoPVCNames,
```

Comment:
📝 If I understand correctly, this (volumes but no dedicated) is the pgBackRest "mode" we intend to remove. If so, we should not worry about optimizing it in this PR.